manifests/slave.pp

```diff
@@ -145,31 +114,110 @@
         before => Service['jenkins-slave'],
       }
     }
+    'Darwin': {
+      $defaults_location = $slave_home
+    }
     default: {
       $defaults_location = '/etc/sysconfig'
     }
   }
 
-  file { '/etc/init.d/jenkins-slave':
-    ensure => 'file',
-    mode   => '0755',
-    owner  => 'root',
-    group  => 'root',
-    source => "puppet:///modules/${module_name}/jenkins-slave.${::osfamily}",
-    notify => Service['jenkins-slave'],
+  case $::kernel {
+    'Linux': {
```

Comment:
This case statement smells a bit to me, would it be possible to refactor this into a `jenkins::slave::linux` and a `jenkins::slave::darwin` module which would then be included depending on the kernel?